### PR TITLE
Free Trials: Create a signup flow for free trials

### LINKED ISSUE #1374: Free Trials: create a signup flow for free trials
Since we won't be offering domains in free trials (p6qnuF-129-p2), we won't be able to send users to the default signup flow. We need to create a specific trial flow that doesn't include the Domain step.

### PR DESCRIPTION
**Pull request by scruffian**
This pull request fixes #1374 by creating a new signup flow that removes the domains step since we don't offer domains as part of free trials.
 
#### Testing instructions: Plans

1. Run `git checkout update/1374-create-free-trial-flow` and start your server
2. Open http://calypso.localhost:3000/plans
3. Check that you are offered to start a free trial on a site that has no plans
4. Start a free trial
5. Assert that a free trial is added to the cart
5. Assert that you are not offered a free trial on /plans
6. Purchase Premium
7. Assert that you are not offered a free trial on /plans

#### Testing instructions 2: Signup without a trial

1. Open http://calypso.localhost:3000/start
2. Move to the plans step
3. Assert that you are not offered a free trial
4. Select a plan
5. Assert that a paid plan is added to the cart

#### Testing instructions 3: Signup with a trial

1. Open http://calypso.localhost:3000/start/free-trial
2. Move to the plans step
3. Assert that you are offered a free trial
4. Select a plan
5. Assert that a free trial is added to the cart

#### Note

This PR also modifies the behaviour of the image button; it will have the same effect as the primary button on the corresponding plan. (If the button says Start Free Trial, the image button should start a free trial).

#### Reviews

- [x] Code
- [x] Product

cc @spncrb @drewblaisdell

### CODE REVIEW

File: client/components/plans/plan-actions/index.jsx [hunk 1]
```diff
@@ -142,6 +140,21 @@ module.exports = React.createClass( {
 		return true;
 	},
 
+	shouldOfferFreeTrial: function() {
+		if ( config.isEnabled( 'upgrades/free-trials' ) ) {
```
Comment (drewblaisdell) [inline client/components/plans/plan-actions/index.jsx]: We can avoid nested conditions here:

```js
if ( ! config.isEnabled( 'upgrades/free-trials' ) ) {
  return false;
}

if ( this.props.isInSignup && this.props.flowName !== 'free-trial' ) {
  return false;
}

if ( this.props.siteSpecificPlansDetails && ! this.props.siteSpecificPlansDetails.can_start_trial ) {
  return false;
}

return true;
```

File: client/signup/steps/plans/index.jsx [hunk 2]
```diff
@@ -86,6 +86,7 @@ module.exports = React.createClass( {
 				<PlanList
 					plans={ this.state.plans }
 					comparePlansUrl={ this.comparePlansUrl() }
+					flowName={ this.props.flowName }
```
Comment (drewblaisdell) [inline client/signup/steps/plans/index.jsx]: We can avoid adding more signup-specific names to `PlanList` and its children by changing this prop name to something like `enableFreeTrials`, and checking the flow name here while we're still in a `client/signup/` file:

```js
enableFreeTrials={ 'free-trials' === this.props.flowName }
```

File: client/components/plans/plan-actions/index.jsx [hunk 3]
```diff
@@ -142,6 +140,21 @@ module.exports = React.createClass( {
...
+			if ( this.props.isInSignup && this.props.flowName !== 'free-trial' ) {
```
Comment (drewblaisdell) [inline client/components/plans/plan-actions/index.jsx]: I think it's a little strange to check one prop only if another prop is true. `this.props` will always be an object in this case, so the `isInSignup` check isn't necessary here.

**Comment by spncrb** [comment]
Issue: Plans:
1) Run git checkout update/1374-create-free-trial-flow and start your server
2) Open http://calypso.localhost:3000/plans
3) Check that you are offered to start a free trial on a site that has no plans
4) Click Start Free trial
5) It is added to your cart, but instead of clicking "Start 14 day free trial", click my sites again
6) Click Plans again
7) Click the cart, which now has 1 item
8) Try to checkout, get this error in console, checkout never works:
screen shot 2015-12-18 at 5 24 02 pm

**Comment by spncrb** [comment]
Testing 2 Works as expected, the only concern I had is removing from the cart returns the user to the plans page, which has free trials enabled - is that not the intention?  If it is, then it :+1:

**Comment by spncrb** [comment]
Test 3 works as expected - waiting on reply for Test 1 issue and test 2 question, then will mark reviewed.

**Comment by Tug** [comment]
@spncrb I can't reproduce the error of your second testing. Could you try again please :)
You can update the branch with:
```
git fetch origin
git checkout update/1374-create-free-trial-flow
git reset --hard origin/update/1374-create-free-trial-flow
```

**Comment by spncrb** [comment]
Thanks @Tug , I still am having this issue.
 Here is the trace:
screen shot 2015-12-21 at 8 56 21 am

File: client/my-sites/plans/main.jsx [hunk 4]
```diff
@@ -89,6 +89,7 @@ module.exports = React.createClass( {
 						sites={ this.props.sites }
 						plans={ this.props.plans.get() }
 						siteSpecificPlansDetailsList={ this.props.siteSpecificPlansDetailsList }
+						enableFreeTrials={ true }
```
Comment (drewblaisdell) [inline client/my-sites/plans/main.jsx]: @Tug thanks for making this change.

Also, not necessarily something we need to change, but there is a shorthand for passing props that are true where you can omit the assignment:

```js
<PlansList enableFreeTrials />
```

**Comment by drewblaisdell** [comment]
> Testing 2 Works as expected, the only concern I had is removing from the cart returns the user to the plans page, which has free trials enabled - is that not the intention?

@spncrb Good catch. This is something we'll deal with when we set up the a/b test.

**Comment by spncrb** [comment]
Was able to reproduce this in /master so it is not PR specific - this is now product review approved.

**Comment by drewblaisdell** [comment]
I updated the previous commit with a typo-fix (`free-trial` instead of `free-trials`). The code looks good to me.

**Pull request merged**